MSAL/test/unit/MSALAuthorityTests.m

```diff
@@ -183,34 +183,44 @@ - (void)testIsKnownHost
     XCTAssertTrue([MSALAuthority isKnownHost:[NSURL URLWithString:@"https://login-us.microsoftonline.com"]]);
 }
 
+- (void)testAuthorityFromCache_whenNilAuthority_shouldFail
+{
+    XCTAssertFalse([MSALAuthority addToValidatedAuthority:nil userPrincipalName:nil]);
+}
 
-- (void)testValidatedAuthorityCache
+- (void)testAuthorityFromCache_whenAdfsWithNilUpn_shouldFail
+{
+    MSALAuthority *adfsAuthority = [MSALTestAuthority ADFSAuthority:[NSURL URLWithString:@"https://fs.contoso.com/adfs/"]];
+    XCTAssertFalse([MSALAuthority addToValidatedAuthority:adfsAuthority userPrincipalName:nil]);
+}
+
+- (void)testAuthorityFromCache_whenCached_shoulRetrieve
 {
     MSALAuthority *aadAuthority = [MSALTestAuthority AADAuthority:[NSURL URLWithString:@"https://login.microsoftonline.in/common"]];
     MSALAuthority *b2cAuthority = [MSALTestAuthority B2CAuthority:[NSURL URLWithString:@"https://login.microsoftonline.in/tfp"]];
     MSALAuthority *adfsAuthority = [MSALTestAuthority ADFSAuthority:[NSURL URLWithString:@"https://fs.contoso.com/adfs/"]];
     
-    // Add non valid authority
-    XCTAssertFalse([MSALAuthority addToValidatedAuthority:nil userPrincipalName:nil]);
-    XCTAssertFalse([MSALAuthority addToValidatedAuthority:adfsAuthority userPrincipalName:nil]);
-    
     // Add valid authority
     XCTAssertTrue([MSALAuthority addToValidatedAuthority:aadAuthority userPrincipalName:nil]);
     XCTAssertTrue([MSALAuthority addToValidatedAuthority:b2cAuthority userPrincipalName:nil]);
     XCTAssertTrue([MSALAuthority addToValidatedAuthority:adfsAuthority userPrincipalName:@"user@contoso.com"]);
 
     // Check if valid authority returned
-    MSALAuthority *retrivedAuthority = [MSALAuthority authorityFromCache:aadAuthority.canonicalAuthority userPrincipalName:nil];
+    MSALAuthority *retrivedAuthority = [MSALAuthority authorityFromCache:aadAuthority.canonicalAuthority
+                                                           authorityType:AADAuthority
+                                                       userPrincipalName:nil];
```

Comment:
It might be a good idea to also add a test case with authorityType == AADAuthority and non-nil userPrincipalName to make sure we test such cases in future too (since it's a valid case for ATS).